### PR TITLE
Remove Rapatch

### PR DESCRIPTION
**Pull request by Maijin**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**

Remove rapatch - untested, not working and dup with other features thus confusing.

### CODE REVIEW

**Review by wargio — APPROVED** [review]

**Comment by ret2libc** [comment]
The cmd_descs files are autogenerated, as you can see from the headers at the top. You should change cmd_descs.yaml

**Review by XVilka — APPROVED** [review]
Please fix the commands YAML, but overall is good. Always good to remove unnecessary code 👍 
We probably should support better existing patching programs or reimplement it from scratch.

File: librz/main/rizin.c [hunk 1]
```diff
@@ -1362,17 +1352,7 @@ RZ_API int rz_main_rizin(int argc, const char **argv) {
...
-		if (data) {
-			ret = rz_core_patch (r, data);
-			rz_core_seek (r, 0, true);
-			free (data);
-		} else {
-			eprintf ("[p] Cannot open '%s'\n", patchfile);
-		}
-	}
-	if ((patchfile && !quiet) || !patchfile) {
+	if (!quiet) {
```
Comment (ret2libc) [inline librz/main/rizin.c]: mmm i think this should be always done.
The condition becomes:
`if ((false && !quiet) || !false)` which is equal to `if (false || true)` -> `if (true)`. Right?

**Review by ret2libc — CHANGES_REQUESTED** [review]
Also, I think you should remove patch.o from the Makefile as well.

**Comment by wargio** [comment]
`make[3]: *** No rule to make target 'patch.o', needed by 'librz_core.so'.  Stop.`

**Review by ret2libc — APPROVED** [review]

**Comment by XVilka** [comment]
```
[XX] db/cmd/cmd_w wp
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'echo 0x10 "hello" > .rz-wp.test
echo 0x20 01020304 >> .rz-wp.test
cat .rz-wp.test
wp .rz-wp.test
rm .rz-wp.test
ps @ 0x10
p8 3 @ 0x20
' -
-- stdout
@@ -1,4 +1,2 @@
 0x10 "hello"
 0x20 01020304
-hello
-010203


-- stderr
Error while executing command: wp .rz-wp.test
```

**Comment by Maijin** [comment]
Current broken tests don't seem related.

**Comment by wargio** [comment]
i'm merging this even if red since the issue is unrelated to this one.

**Comment by ret2libc** [comment]
Yeah it's very likely just MacOS on maintenance on GHA.

**Pull request merged**